README.md

```diff
@@ -0,0 +1,205 @@
+
+### Activating the builder
+
+You'll need to activate `storybook-builder-wds` manually.
+To do that change the following in `main.js`:
+
+- set `config` type to `import('storybook-builder-wds').StorybookConfigWds`
+- set the `core.builder` setting to `'storybook-builder-wds'`
+- keep the `framework.name` (Vite won't be used as a builder, but the chosen framework, e.g. `web-components`, will be inherited from this setting)
+
+```js
+// .storybook/main.js
+
+/** @type { import('storybook-builder-wds').StorybookConfigWds } */
+const config = {
+  ...
+  framework: {
+    name: '@storybook/web-components-vite',
+  },
+  core: {
+    builder: 'storybook-builder-wds',
+  },
+  ...
+};
+
+export default config;
+```
+
+### Configuring storybook
+
+The `storybook-builder-wds` aims at compatibility with standard storybook features, e.g. `.storybook/preview.js`.
+Follow the [official storybook docs](https://storybook.js.org/) for the configuration options.
+
+### Configuring @web/dev-server
+
+The builder comes with a default `@web/dev-server` configuration which adheres to browser standards and supports essential plugins of the storybook ecosystem.
+
+When storybook is loaded, the default config gets automatically merged with the project configuration file (`web-dev-server.config.{mjs,cjs,js}`) if present.
+This is needed to ensure that the same configuration is used for application, feature or design system you are building.
+
+#### wdsFinal hook
+
+Sometimes you might need to add a storybook specific configuration.
+There is `wdsFinal` hook for that.
+
+```js
+// .storybook/main.js
+import { mergeConfigs } from '@web/dev-server';
+
+/** @type { import('storybook-builder-wds').StorybookConfigWds } */
+const config = {
+  ...
+  core: {
+    builder: 'storybook-builder-wds',
+  },
+  async wdsFinal(config) {
+    return mergeConfigs(config, {
+      // add storybook specific configuration for @web/dev-server
+      // e.g. "open" to go to a custom URL in the browser when server has started
+      open: '/custom-path',
+    });
+  },
+  ...
+};
+
+export default config;
+```
+
+## Migration
+
+This guide assumes you are migrating from the `@web/dev-server-storybook` plugin and `@web/storybook-prebuilt` storybook 6 bundle.
+
+Most of the [official migration guide](https://storybook.js.org/docs/web-components/migration-guide) applies to this migration too, with a few additions and exceptions.
+Please read the official guide and follow it for manual migrations, but before running the automatic `upgrade` script read the sections below.
+
+### Run upgrade script
+
+The automatic `upgrade` script can't detect the `@web/dev-server-storybook` plugin and won't work as a result.
+But with a simple workaround you can make it work.
+
+First install old version of `builder-vite` and old storybook 6 renderer for the technology you used in the old setup, e.g. for Web Components it will be:
```

Comment:
Is this _really_ the easiest way to upgrade?

Reply:
this is a workaround, not ideal, but same trade-off as with the setup script